ignite/engine/engine.py

```diff
@@ -994,7 +999,11 @@ def _internal_run_as_gen(self) -> Generator[Any, None, State]:
             # time is available for handlers but must be updated after fire
             self.state.times[Events.COMPLETED.name] = time_taken
             handlers_start_time = time.time()
-            self._fire_event(Events.COMPLETED)
+
+            # do not fire Events.COMPLETED if we terminated the run with flag `skip_completed_after_termination=True`
+            if self.should_terminate and not self.skip_completed_after_termination or not self.should_terminate:
+                self._fire_event(Events.COMPLETED)
```

Comment:
I think this condition should be:
```suggestion
            if not (self.should_terminate and self.skip_completed_after_termination):
                self._fire_event(Events.COMPLETED)
```

Maybe, all the code below that updates `self.state.times[Events.COMPLETED.name] = time_taken` can be inside the if clause

Reply:
Agree on the reformulation of the if condition. 

Concerning your second point, could this create compatibility issues with existing handlers accessing `self.state.times[Events.COMPLETED.name]`? If we move the code below line 1100 into the if, in case of `Events.TERMINATE` the state `self.state.times[Events.COMPLETED.name]` would not be written...